pip/basecommand.py

```diff
@@ -314,6 +314,23 @@ def populate_requirement_set(requirement_set, args, options, finder,
                     'You must give at least one requirement to %(name)s '
                     '(see "pip help %(name)s")' % opts)
 
+        # On Windows, any operation modifying pip should be run as:
+        #     python -m pip ...
+        # See https://github.com/pypa/pip/issues/1299 for more discussion
+        should_show_use_python_msg = (
+            sys.platform == 'win32' and
```

Comment:
Both Jython and IronPython have `sys.platform` attribute, but its value is different to CPython. On Jython it's something like `'java1.8.0_131'` and on IronPython `'cli'`. These values don't change depending on the OS.

I agree this is a topic where comments from core developers would be good. Preventing changes to pip breaking it on alternative Python implementations would probably also require some collaboration with the developers of these implementations.